internal/attestation/attestation.go

```diff
@@ -0,0 +1,234 @@
+		// Generate and persist the AK
+		if err := GenerateAndPersistAK(logger); err != nil {
+			return err
+		}
+
+		attestationKeyBytes, err := os.ReadFile(AKPath)
+		if err != nil {
+			return err
+		}
+		attestationKey := string(attestationKeyBytes)
+
+		// Check if the neednet file exists to determine our retry behavior
+		_, needNetErr := os.Stat(needNetPath)
+		needNetExists := (needNetErr == nil)
+		if needNetExists {
+			logger.Info("neednet file exists, network should be available for attestation")
+		} else {
+			logger.Info("neednet file does not exist, will return ErrNeedNet if network is unavailable")
+		}
+
+		err = AttestationKeyRegistration(logger, cfg.Attestation.AttestationKey.Registration,
+			attestationKey, platformName)
+		if err != nil {
+			// If we got ErrNeedNet and the neednet file doesn't exist, propagate it
+			// (we're in fetch-offline and need to signal for network)
+			if err == resource.ErrNeedNet && !needNetExists {
+				return err
+			}
+			// If we got ErrNeedNet but neednet file exists, we're in fetch stage
+			// Retry the registration with delays to allow network to come up
+			if err == resource.ErrNeedNet && needNetExists {
+				logger.Info("Network not ready yet in fetch stage, retrying with delays...")
+				// Retry up to 10 times with increasing delays
+				maxRetries := 20
+				for attempt := 2; attempt <= maxRetries; attempt++ {
+					delay := time.Duration(min(attempt*2, 10)) * time.Second
+					logger.Info("Waiting %v before retry attempt %d/%d", delay, attempt, maxRetries)
+					time.Sleep(delay)
+
+					err = AttestationKeyRegistration(logger, cfg.Attestation.AttestationKey.Registration,
+						attestationKey, platformName)
+					if err == nil {
+						break
+					}
+					logger.Info("Attestation registration attempt %d/%d failed: %v", attempt, maxRetries, err)
+				}
+				if err != nil {
+					return fmt.Errorf("failed to register attestation key after retries: %w", err)
+				}
+			} else {
+				return err
+			}
+		}
+	}
+	return nil
+}
+
+// GenerateAndPersistAK creates and persists the Attestation Key in the TPM
+func GenerateAndPersistAK(logger *log.Logger) error {
+	if err := os.MkdirAll(TPMDir, 0755); err != nil {
+		return fmt.Errorf("couldn't create %s directory: %w", TPMDir, err)
+	}
+
+	if _, err := os.Stat(AKPath); err == nil {
+		logger.Info("Attestation Key already exists, skipping generation")
+		return nil
+	}
+
+	logger.Info("Generating Attestation Key")
+	cmd := exec.Command("tpm2_createak", "-C", EKHandle,
+		"-c", AKCtxPath, "-G", "rsa", "-g", "sha256",
+		"-s", "rsassa", "-u", AKPath, "-f", "pem")
+	if _, err := logger.LogCmd(cmd, "creating attestation key"); err != nil {
+		return fmt.Errorf("failed to create attestation key: %w", err)
+	}
+
+	cmd = exec.Command("tpm2_evictcontrol", "-c", AKCtxPath, AKHandle)
+	if _, err := logger.LogCmd(cmd, "persisting attestation key"); err != nil {
+		return fmt.Errorf("failed to persist attestation key: %w", err)
+	}
```

Comment:
This commend assume ```EKHandle``` exist but it is not always the case (e.g on GCP). I suggest
```suggestion
	cmd := exec.Command("tpm2_createprimary", "-C", "e",
		"-c", AKCtxPath, "-G", "rsa2048:rsassa:null", "-g", "sha256","-a", 
		"fixedtpm|fixedparent|sensitivedataorigin|userwithauth|restricted|sign")
	if _, err := logger.LogCmd(cmd, "creating attestation key"); err != nil {
		return fmt.Errorf("failed to create attestation key: %w", err)
	}

	cmd = exec.Command("tpm2_evictcontrol","-C", "o", "-c", AKCtxPath, AKHandle)
	if _, err := logger.LogCmd(cmd, "persisting attestation key"); err != nil {
		return fmt.Errorf("failed to persist attestation key: %w", err)
	}
	cmd = exec.Command("tpm2_readpublic", "-c", AKHandle, "-f", "pem", "-o", AKPath)
	if _, err := logger.LogCmd(cmd, "exporting attestation key public part"); err != nil {
		return fmt.Errorf("failed to export attestation key public part: %w", err)
	}
```
This will create AK under the endorsement hierarchy.